src/runner.rs

```diff
@@ -139,9 +144,57 @@ fn parse_test(test_path: &Path, config: &TestConfig) -> InnerTestResult<Test> {
         expected_stdout,
         expected_stderr,
         expected_exit_status,
+        rest,
     })
 }
 
+fn overwrite_test(test_path: &PathBuf, config: &TestConfig, output: &Output, test: &Test) -> std::io::Result<()> {
+    // Maybe copy the file so we don't remove it if we fail here?
+    let mut file = File::create(test_path)?;
+
+    file.write_all(test.rest.as_bytes())?;
+
+    if !test.command_line_args.is_empty() {
+        writeln!(file, "{}{}", config.test_args_prefix, test.command_line_args)?;
+    }
+
+    if Some(0) != output.status.code() {
+        writeln!(
+            file,
+            "{} {}",
+            config.test_exit_status_prefix,
+            output.status.code().unwrap_or(0)
+        )?;
+    }
+
+    // Doesn't handle \r correctly!
+    if output.stdout.len() != 0 {
+        writeln!(file, "{}", config.test_stdout_prefix)?;
+        for line in output.stdout.split(|c| *c == 0x0A) {
+            if line.is_empty() {
+                continue;
```

Comment:
Can you elaborate on why this check is needed? It seems to me it could cause tests to fail if we don't write empty lines and the output contains possibly multiple empty lines in the middle.

Reply:
They might be an artifact of me writing this in the middle of the night. I read the documentation (https://doc.rust-lang.org/std/primitive.str.html#method.split), as it's possible to get an empty slice from the split in the loop above. 
```
let x = "||||a||b|c".to_string();
let d: Vec<_> = x.split('|').collect();

assert_eq!(d, &["", "", "", "", "a", "", "b", "c"]);
```
But we probably want to write the line endings here, so I think your hunch is right. I've removed them.

Reply:
Thanks - an empty slice returned from split can be perfectly expected since the slices do not contain the separator character. So if your string is "aaa" and you spilt on 'a' your slices would be `["", "", "", ""]`, but when re-writing to the output as we do here you want to rewrite that separator character even after empty slices instead of skipping them. In our case it looks like this causes us to output an extra newline at the end, though luckily since it is just whitespace we can trim it.